3744: Rescue module for OpenList

martinb14 opened the pull request:
#### Link to issue

https://platform.dandigbib.org/issues/3744

#### Description

A module for rescuing certain lists and elements

cableman commented:
Please fill out the template given when creating new pull request (https://github.com/ding2/ding2/blob/master/.github/PULL_REQUEST_TEMPLATE.md). 

Please also look at the steps 3,4 and 5 in contributing file that where agreed upon when making this PR's to the project. See https://github.com/ding2/ding2/blob/master/.github/CONTRIBUTING.md

These guidelines exists to make the process of contributing to the project as fast and easy for all involved in the process and to give others an easy view of the changes for a given PR.

Please follow these guidelines.

martinb14 commented:
Link to issue
https://platform.dandigbib.org/issues/3744

Description
A module for rescuing certain lists and elements

Screenshot of the result
![image](https://user-images.githubusercontent.com/1502991/47000811-584b6300-d129-11e8-860d-1c1c5b2b8292.png)


Checklist
YES: My complies with our coding guidelines.
YES: My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
YES: My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
![image](https://user-images.githubusercontent.com/1502991/47000658-04d91500-d129-11e8-93c2-d135d2bbac9d.png)

Additional comments or questions
Im not sure I understand the checklist above. Isn't this what github does for us? Or does this imply that we have to run other test suites on our side?

cableman commented:
If you use the template and [x] for each check we can see in the overview that you have locked through you code and check that it complies with the guidelines.

9/10 times both Scrutinizer and CI has not passed the checks for PR's and the core team then needs to go through these which is time consuming (when having 200+ PR's) and slows down the release process. May issues are return to the developer, just because the tests are not passed.

So used correctly we will get a much better overview (see the out-of 3/3 in the overview):

![screen shot 2018-10-17 at 09 38 09](https://user-images.githubusercontent.com/111397/47069931-6dd79000-d1f0-11e8-8265-bee3515dad5d.png)

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.info:
```diff
@@ -0,0 +1,5 @@
+description = Recreate missing lists and elements.
+core = 7.x
+package = Ding!
+dependencies[] = ding_list
```
Missing new line at the end of the file (see the red icon).

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+  $form_state['rebuild'] = TRUE;
+
+  // Load the user object.
+  $account = user_load($user->uid);
```
Why load the user account when you have the user object in $user?

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+/**
+ * Logging of actions
+ */
+function ding_listrescue_log($msg) {
```
Not sure why this logging with drupal set message is need?

cableman replied:
If required function. The function documentation is missing @param

cableman replied:
Please see the first question? Why this debug statement is here?

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+ * Get local elements from a local list.
+ */
+function ding_listrescue_get_local_elements($local) {
+  $max_elements = 200;
```
Why the limit of 200?

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+ * Get all the local lists.
+ */
+function ding_listrescue_get_local_lists($account = NULL, $types = NULL) {
+  $max_lists = 50;
```
Why the limit of 50?

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+  if (empty($account)) {
+    global $user;
+    $account = user_load($user->uid);    
+  }
```
Why the user load? The $user is the same as what you get from user_load?

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+
+/**
+ * Get local elements from a local list.
+ */
```
Missing @param, @return function documentation.

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+/**
+ * Get all the local lists.
+ */
+function ding_listrescue_get_local_lists($account = NULL, $types = NULL) {
```
Missing @param, @return function documentation.

cableman commented on modules/p2/ding_list/modules/ding_listrescue/ding_listrescue.module:
```diff
@@ -0,0 +1,240 @@
+
+/**
+ * Get a matching remote list.
+ */
```
Missing @param, @return function documentation.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 57497e51762f90fa0f8c78632b37c806a2e97208
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 57497e51762f90fa0f8c78632b37c806a2e97208
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 4a9ee51fb512fea63d5081646421c152d7fa313a
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 4a9ee51fb512fea63d5081646421c152d7fa313a
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

martinb14 commented:
Fixed @cableman change requests

cableman commented:
Please answer the question to ensure at each comment to ensure that you actually have seen them all and to speed up the review process.

Summery (questions now answered):
1. Why the debug logger message function?
2. Why the extra user_loads, when you already have the global user object?
3. Stil why the limit (setting them higher do not answer the question)?

martinb14 commented:
1) Logger removed
2) user_load removed (Previous bug in Drupal required this in some cases)
3) Comments describes why the limit.
